searx/engines/arxiv.py

```diff
@@ -27,6 +28,23 @@
 # engine dependent config
 number_of_results = 10
 
+# xpaths
+arxiv_namespaces = {
+    "atom": "http://www.w3.org/2005/Atom",
+    "arxiv": "http://arxiv.org/schemas/atom",
+}
+xpath_entry = XPath('//atom:entry', namespaces=arxiv_namespaces)
+xpath_title = XPath('.//atom:title', namespaces=arxiv_namespaces)
+xpath_id = XPath('.//atom:id', namespaces=arxiv_namespaces)
+xpath_summary = XPath('.//atom:summary', namespaces=arxiv_namespaces)
+xpath_author_name = XPath('.//atom:author/atom:name', namespaces=arxiv_namespaces)
+xpath_doi = XPath('.//arxiv:doi', namespaces=arxiv_namespaces)
+xpath_pdf = XPath('.//atom:link[@title="pdf"]', namespaces=arxiv_namespaces)
+xpath_published = XPath('.//atom:published', namespaces=arxiv_namespaces)
+xpath_journal = XPath('.//arxiv:journal_ref', namespaces=arxiv_namespaces)
+xpath_category = XPath('.//atom:category/@term', namespaces=arxiv_namespaces)
+xpath_comment = XPath('./arxiv:comment', namespaces=arxiv_namespaces)
```

Comment:
Yes me too, except the namespaces must be specified.

Currently (I pick two lines among others):
```python
for entry in eval_xpath_list(dom, xpath_entry):
    title = eval_xpath_getindex(entry, xpath_title, 0).
```

Usually we have:
```python
for entry in eval_xpath_list(dom, '//atom:entry'):
    title = eval_xpath_getindex(entry, './/atom:title', 0).
```

Except it won't work because we have to specify the namespaces.
However the `eval_xpath_*` functions do not accept the `namespaces` parameter. But we can pass a XPath instance instead of a string, so the code would be:
```python
for entry in eval_xpath_list(dom, XPath('//atom:entry', namespaces=arxiv_namespaces)):
    title = eval_xpath_getindex(entry, XPath('.//atom:title', namespaces=arxiv_namespaces), 0).
```

IMO it is less readable compare to the current version.